### ISSUE TITLE
ViMode Bug: System.ArgumentOutOfRangeException: Index was out of range

### ISSUE THREAD

**Issue by jm33-m0** [PowerShell/PSReadLine#1408]
```
Oops, something went wrong.  Please report this bug with the details below.
Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new
-----------------------------------------------------------------------
Last 114 Keys:
 i s i s s s Ctrl+Backspace g e t - p r o Tab Enter
 UpArrow Escape b b b w w 0 $ B $ b b b d w d w d w a a a a a Backspace Backspace Backspace Backspace Backspace Backspace Escape Escape s l s k s c s Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace Backspace c Enter
 l s Enter
 c Enter
 Ctrl+r g t Backspace e t - p s Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Escape Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace g e t - p r Backspace s Tab Ctrl+Backspace Ctrl+Backspace Ctrl+Backspace Ctrl+r Escape Escape Ctrl+s Escape g e t - p Backspace s g

Exception:
System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index
   at System.Text.StringBuilder.Insert(Int32 index, Char* value, Int32 valueCount)
   at Microsoft.PowerShell.PSConsoleReadLine.Insert(Char c)
   at Microsoft.PowerShell.PSConsoleReadLine.SelfInsert(Nullable`1 key, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
   at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
   at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)
-----------------------------------------------------------------------
```

**Comment by iSazonov** [comment]
See #1306

**Comment by jm33-m0** [comment]
@iSazonov I am using the lastest version which has #1306 fixed, so I guess #1408 is a different bug. BTW I found it when randomly hitting keyboard under ViMode

**Comment by daxian-dbw** [comment]
From the bug report template:
```
The maintainer may close your issue without further explanation or engagement if:
- You delete this entire template and go your own path;
```
So, please follow the bug report template to provide **environment data** and **repro steps** the next time.

For this issue, are you sure you have the latest version of PSReadLine? The url `Report on GitHub: https://github.com/lzybkr/PSReadLine/issues/new` indicates you were using a fairly old version.
At any rate, this is a duplicate of #1258 and was fixed in the latest version.
